native/single_precision

```diff
@@ -0,0 +1,149 @@
+      -e 's_^// D_// S_' \
+>> level1single_sdot.go
+
+echo Generating level1single_dsdot.go
+echo -e '// Generated code do not edit. Run `go generate`.\n' > level1single_dsdot.go
+cat level1double_ddot.go \
+| gofmt -r '[]float64 -> []float32' \
+\
+| gofmt -r 'asm.DdotInc -> asm.DsdotInc' \
+| gofmt -r 'asm.DdotUnitary -> asm.DsdotUnitary' \
+\
+| sed -e 's_^\(func (Implementation) \)D\(.*\)$_//\n// Float32 implementations are autogenerated and not directly tested.\n\1Ds\2_' \
+      -e 's_^// D_// Ds_' \
+>> level1single_dsdot.go
+
+echo Generating level1single_sdsdot.go
+echo -e '// Generated code do not edit. Run `go generate`.\n' > level1single_sdsdot.go
+cat level1double_ddot.go \
+| gofmt -r 'float64 -> float32' \
+\
+| gofmt -r 'asm.DdotInc(x, y, f(n), f(incX), f(incY), f(ix), f(iy)) -> alpha + float32(asm.DsdotInc(x, y, f(n), f(incX), f(incY), f(ix), f(iy)))' \
+| gofmt -r 'asm.DdotUnitary(a, b) -> alpha + float32(asm.DsdotUnitary(a, b))' \
+\
+| sed -e 's_^\(func (Implementation) \)D\(.*\)$_//\n// Float32 implementations are autogenerated and not directly tested.\n\1Sds\2_' \
+      -e 's_^// D\(.*\)$_// Sds\1 plus a constant_' \
+      -e 's_\\sum_alpha + \\sum_' \
+      -e 's/n int/n int, alpha float32/' \
+>> level1single_sdsdot.go
+
+
+# Level2 routines.
+
+echo Generating level2single.go
+echo -e '// Generated code do not edit. Run `go generate`.\n' > level2single.go
+cat level2double.go \
+| gofmt -r 'blas.Float64Level2 -> blas.Float32Level2' \
+\
+| gofmt -r 'float64 -> float32' \
+\
+| gofmt -r '0.0 -> float32(0.0)' \
+| gofmt -r '1.0 -> float32(1.0)' \
+\
+| gofmt -r 'Dscal -> Sscal' \
+\
+| gofmt -r 'math.Abs(a) -> float32(math.Abs(float64(a)))' \
+| gofmt -r 'math.Copysign(a, b) -> float32(math.Copysign(float64(a), float64(b)))' \
+| gofmt -r 'math.Hypot(a, b) -> float32(math.Hypot(float64(a), float64(b)))' \
+| gofmt -r 'math.Sqrt(a) -> float32(math.Sqrt(float64(a)))' \
+\
+| gofmt -r 'asm.DaxpyInc -> asm.SaxpyInc' \
+| gofmt -r 'asm.DaxpyUnitary -> asm.SaxpyUnitary' \
+| gofmt -r 'asm.DdotInc -> asm.SdotInc' \
+| gofmt -r 'asm.DdotUnitary -> asm.SdotUnitary' \
+\
+| sed -e 's_^\(func (Implementation) \)D\(.*\)$_//\n// Float32 implementations are autogenerated and not directly tested.\n\1S\2_' \
+      -e 's_^// D_// S_' \
+>> level2single.go
+
+
+# Level3 routines.
+
+echo Generating level3single.go
+echo -e '// Generated code do not edit. Run `go generate`.\n' > level3single.go
+cat level3double.go \
+| gofmt -r 'blas.Float64Level3 -> blas.Float32Level3' \
+\
+| gofmt -r 'float64 -> float32' \
+\
+| gofmt -r 'asm.DaxpyUnitary -> asm.SaxpyUnitary' \
+| gofmt -r 'asm.DdotUnitary -> asm.SdotUnitary' \
+\
+| sed -e 's_^\(func (Implementation) \)D\(.*\)$_//\n// Float32 implementations are autogenerated and not directly tested.\n\1S\2_' \
+      -e 's_^// D_// S_' \
+>> level3single.go
+
+echo Generating general_single.go
+echo -e '// Generated code do not edit. Run `go generate`.\n' > general_single.go
+cat general.go \
```

Comment:
I think general.go should be renamed to general_double.go to be consistent. Same for general to general64 and newGeneral to newGeneral64.